building: retry mechanism: enable retry on winerror 110

rokm opened the pull request:
Add winerror 110 (`ERROR_OPEN_FAILED`) to the list of permissible error codes for the retry mechanism that attempts to mitigate build failures due to anti-virus program interference.

At the same time, reorganize the exception-checking code by merging all Windows specific code into a single branch. This allows us to re-use `_ALLOWED_WINERROR` with both `pywintypes.error` and `OSError`. With the latter, we now compare the `winerror` attribute as a fall back after `errno` comparison.

Closes #8138.

rokm committed:
building: retry mechanism: enable retry on winerror 110

Add winerror 110 (`ERROR_OPEN_FAILED`) to the list of permissible
error codes for the retry mechanism that attempts to mitigate
build failures due to anti-virus program interference.

At the same time, reorganize the exception-checking code by
merging all Windows specific code into a single branch. This allows
us to re-use `_ALLOWED_WINERROR` with both `pywintypes.error` and
`OSError`. With the latter, we now compare the `winerror` attribute
as a fall back after `errno` comparison.

bwoodsend commented:
Is `tests/requirements-libraries.txt` missing a pin somewhere? Shouldn't these random Qt regressions only occur during PyUp updates?

rokm commented:
> Is `tests/requirements-libraries.txt` missing a pin somewhere? Shouldn't these random Qt regressions only occur during PyUp updates?

We're not pinning the `-Qt5` / `-Qt6` packages of PyQt5/6 (nor the PySide6-Essentials and PySide6-AddOns for `PySide6`). We used to have the PyQt5-Qt5 and PyQt6-Qt6 pinned at some point, but I removed those when I added additional PyQt packages (PyQtCharts, PyQt6-QtCharts, etc.), because each of those would also need their own `-Qt5/-Qt6` package pinned. 

Think I'll re-pin those, though, as well as add pins for all other `-Qt5/-Qt6` packages. It's going to be ugly, but PyUp should manage...
